streams/src/main/java/org/apache/kafka/streams/state/internals/StreamThreadStateStoreProvider.java

```
@@ -40,6 +41,9 @@ public StreamThreadStateStoreProvider(final StreamThread streamThread) {
     @SuppressWarnings("unchecked")
     @Override
     public <T> List<T> stores(final String storeName, final QueryableStoreType<T> queryableStoreType) {
+        if (streamThread.state() == StreamThread.State.DEAD) {
```

I'm wondering if we can change the name of the method `StreamThread#isInitialized()` to something else a bit more meaningful and then that method could become:
`return state == State.RUNNING || state == State.DEAD`

We then don't need the `if(streamThread.state() == StreamThread.State.DEAD)` as `streamThread.tasks()` will be empty so we'll return an empty `List`

---

I see your point, but I am not sure if this would make the code too convoluted. Basically, we can only query in state `RUNNING`, want to ignore in state `DEAD` and want to fail for the other 3 states. Right now, the code is quite explicit about this.

If you insist, we could got with `StreamThread#isQueryable()` but we would need a comment why a DEAD thread is queryable. Thoughts?

---

I'm wondering why we want to treat `Dead` with `PARTITIONS_REVOKED | etc` differently? I think in both cases we can throw an exception to let the caller re-discover the state stores since they should be migrated to other threads right?

---

That is exactly the thing we need to distinguish (ie, that is the state before this change) -- after a thread dies and the stores are migrated, the `StreamThreadStoreSupplier` of the died thread will still be checked if it can be queries -- thus, if we throw the whole attempt to query any store will fail, as a store is only queryable if all shards are queryable. Thus, a dead thread must not fail to make a store queryable.

---

@mjsax - leave it as it is. A `Dead` thread isn't queryable. So that would be a bit confusing